wren-ui/src/apollo/server/services/askingService.ts

```diff
@@ -342,6 +372,67 @@ export class AskingService implements IAskingService {
     return threadResponse;
   }
 
+  public async createCorrectedThreadResponse(
+    threadId: number,
+    input: CorrectedDetailTaskInput,
+  ): Promise<ThreadResponse> {
+    const thread = await this.threadRepository.findOneBy({
+      id: threadId,
+    });
+
+    if (!thread) {
+      throw new Error(`Thread ${threadId} not found`);
+    }
+
+    const baseThreadResponse = await this.threadResponseRepository.findOneBy({
```

Comment:
(NIT) 
I think there is no need to check the existence of a thread  while checking threadResponse is needed.
The threadId can be found in the threadResponse record, so it doesn't need to be passed from the client side.